tests/com/twitter/intellij/pants/util/PantsUtilTest.java

```diff
@@ -57,6 +66,38 @@ public void testFindJdk() {
     assertEquals(twoEntriesSameSdk, getSameJdks(sdkA));
   }
 
+  protected String getPantsExecutablePath() {
+    return PantsUtil.findPantsExecutable(getParentPath()).get().getPath();
+  }
+
+  public void testNonUtf8LocaleSubprocessFails() throws ExecutionException {
+    final GeneralCommandLine command = PantsUtil.defaultCommandLine(getPantsExecutablePath());
+    command.addParameter("options");
+
+    final Map<String, String> localeOverrides = new HashMap<>();
+    final String badLocale = "en_US.US-ASCII";
+    localeOverrides.put("LANG", badLocale);
+    localeOverrides.put("LC_ALL", badLocale);
+
+    final Map<String, String> cmdEnv = command.getEnvironment();
+    cmdEnv.putAll(localeOverrides);
+
+    final ProcessOutput processOutput = PantsUtil.getCmdOutput(command, null);
+    assertFalse(processOutput.checkSuccess(LOG));
+
+    final String stderrString = processOutput.getStderr();
+    assertContainsSubstring(stderrString,
+                            "pants.bin.pants_loader.InvalidLocaleError: System preferred encoding is `US-ASCII`, but `UTF-8` is required.");
+  }
+
+  public void testSubprocessWithShellEnvSucceeds() throws ExecutionException {
+    final GeneralCommandLine command = PantsUtil.defaultCommandLine(getPantsExecutablePath());
+    command.addParameter("options");
+
+    final ProcessOutput processOutput = PantsUtil.getCmdOutput(command, null);
+    assertTrue(processOutput.checkSuccess(LOG));
+  }
+
```

Comment:
The tests look good and should stay, but they are not specifically testing the case where intellij was launched with the bad `LANG` or `LC_ALL` env values and then the pants invocation still works, in case that was what you intended to test.

Reply:
That is exactly what I intended to test! I guess an integration test could do this exactly — wasn’t thinking too hard about it before.

Reply:
Sounds good. This is fine as is, but feel free to add more tests.